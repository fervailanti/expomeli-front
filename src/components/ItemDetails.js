import React from 'react'
import ItemDetailSection from './ItemDetailSection'
import ItemSaleDetails from './ItemSaleDetails'
import { formatPrice } from '../utils/formatPrice'
import { formatCondition } from '../utils/formatCondition'
import { WithLoader } from './WithLoader'

const ItemDetails = ({ item }) => {
  const { picture, condition, sold_quantity, title, price, description } = item
  return (
    <div className="product">
      <ItemSaleDetails
        image={picture}
        condition={formatCondition(condition)}
        soldQty={sold_quantity}
        title={title}
        price={formatPrice(price)}
      />
      <ItemDetailSection title="DescripciÃ³n del producto" description={description} />
    </div>
  )
}

export default WithLoader(ItemDetails)
