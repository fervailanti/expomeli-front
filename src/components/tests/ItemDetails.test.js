import React from 'react'
import ItemDetails from '../ItemDetails'
import { render } from '@testing-library/react'

const itemMock = {
  id: 'TESTID1234',
  title: 'iPhone 11 64 Gb',
  price: {
    currency: 'ARS',
    amount: 174999,
    decimals: 99
  },
  picture: 'https://picsum.photos/500',
  condition: 'new',
  free_shipping: true,
  sold_quantity: 13,
  description: 'lorem ipsum'
}

it('ItemDetails component should render correctly', () => {
  const tree = render(<ItemDetails item={itemMock} onBack={() => null} />)
  expect(tree.container).toMatchSnapshot()
})
