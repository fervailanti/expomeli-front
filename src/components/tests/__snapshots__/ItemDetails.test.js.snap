// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetails component should render correctly 1`] = `
<div>
  <button
    class="flex items-center font-thin mb-3 lg:mb-5 hover:text-blue"
  >
    <img
      alt="go-back-arrow"
      class="h-4 mr-2"
      src="[object Object]"
    />
    Volver al listado
  </button>
  <div
    class="product"
  >
    <div
      class="sale"
    >
      <img
        alt="product"
        class="product-img"
        src="https://picsum.photos/500"
      />
      <div
        class="details"
      >
        <div>
          <p
            class="hairline"
          >
            Nuevo | 13 vendidos
          </p>
          <p
            class="title"
          >
            iPhone 11 64 Gb
          </p>
          <p
            class="price"
          >
            $ 174.999,99
          </p>
        </div>
        <div
          class="flex flex-col items-center mb-8"
        >
          <button
            class="btn mt-4"
          >
            Comprar ahora
          </button>
          <button
            class="btn-ghost mt-4"
          >
            Agregar al carrito
          </button>
        </div>
      </div>
    </div>
    <div
      class="detail-section"
    >
      <p
        class="title"
      >
        Descripci√≥n del producto
      </p>
      <p
        class="description"
      >
        lorem ipsum
      </p>
    </div>
  </div>
</div>
`;
